test/e2e-common.sh

```diff
@@ -106,24 +110,32 @@ function pubsub_setup() {
 }
 
 # Create resources required for GCP Broker authentication setup.
-function gcp_broker_setup() {
+function broker_auth_setup() {
   echo "Authentication setup for GCP Broker"
   local auth_mode=${1}
 
   if [ "${auth_mode}" == "secret" ]; then
-    kubectl -n "${CONTROL_PLANE_NAMESPACE}" create secret generic "${GCP_BROKER_SECRET_NAME}" --from-file=key.json="${PUBSUB_SERVICE_ACCOUNT_KEY_TEMP}"
+    if (( ! IS_PROW )); then
+      init_pubsub_service_account "${E2E_PROJECT_ID}" "${BROKER_GSA_NON_PROW}"
```

Comment:
Perhaps this function should be renamed as well?

Reply:
Now that we use a different GSA for the broker we have to create it and grant the right permissions, just similar to the `sources_auth_setup`. I can add a comment similar to the one there:

```# When not running on Prow we need to set up a service account for broker.```

Reply:
I was think of renaming to `init_gsa_with_pubsub_editor` but @zhongduo had a fair point (https://github.com/google/knative-gcp/pull/2089#discussion_r564623456) to keep the old name as the new one will be a bit confusing if later we decided not to use editor role. Do you think of a specific name? This is used both the setup the sources & broker GSA.

Reply:
Just the clarify, my previous comment was not about "pubsub" or not since the new name was using pubsub too. I was saying that too much implementation details in the name is not a good idea, in this case "editor" role.

Reply:
Right, I guess that's what I understood too. I am happy to change this if there is a better name.